### ISSUE TITLE
"rule UCSC_GTF" not compatible with Ensembl GTF

### ISSUE THREAD

**Issue by tmari** [jtpoirier/proteomegenerator#7]
I was trying proteomegenerator on my data, but the analysis was giving me an error at the level of "UCSC_GTF". Looking at the code in pgm

`shell: "cat {GTF} | grep chr > {output.reference}; \`
`    cat {input} | grep chr > {output.merged} 2> {log}"`

but GTF files don't necessarily have "chr" prefix in the chromosome name field (source: [https://www.ensembl.org/info/website/upload/gff.html](url)).

Is the purpose of USCS_GTF to just remove the first lines starting with an hash from the files, or also to remove scaffolds from the GTFs and keep only chromosomal positions? Both are easy fixes but quite different.

**Comment by jtpoirier** [comment]
You are correct – the current version of ProteomeGenerator is not ALT aware but this would be a cool feature to add and we would be happy to test a pull request.

On Nov 15, 2019, at 9:55 AM, tmari <notifications@github.com<mailto:notifications@github.com>> wrote:


I was trying proteomegenerator on my data, but the analysis was giving me an error at the level of "UCSC_GTF". Looking at the code in pgm

shell: "cat {GTF} | grep chr > {output.reference}; \
cat {input} | grep chr > {output.merged} 2> {log}"

but GTF files don't necessarily have "chr" prefix in the chromosome name field (source: https://www.ensembl.org/info/website/upload/gff.html<x-msg://64/url>).

Is the purpose of USCS_GTF to just remove the first lines starting with an hash from the files, or also to remove also scaffolds from the GTFs and keep only chromosomal positions? Both are easy fixes but quite different.
